test/db/formats/demangling_bin

```diff
@@ -176,27 +174,24 @@ pd 1
 EOF
 EXPECT=<<EOF
             ; CALL XREF from sym._init @ 0x8049008
-            ; CALL XREF from QtPrivate::RefCount::deref() @ 0x8049349 ; method.QtPrivate::RefCount.deref
-            ; CALL XREF from QString::QString(char const*) @ 0x804939d ; method.QString.QString_char_const
-            ; CALL XREF from QString::~QString() @ 0x80493e5 ; method.QString._QString
+            ; CALL XREF from method.QtPrivate::RefCount.deref @ 0x8049349
+            ; CALL XREF from method.QString.QString_char_const @ 0x804939d
+            ; CALL XREF from method.QString._QString @ 0x80493e5
             ; CALL XREF from sym._fini @ 0x80495d4
 / sym.__x86.get_pc_thunk.bx ();
 |           0x08049140      mov   ebx, dword [esp]
 
-.--------------------------------------------------------------.
-|  0x8049140                                                   |
-|   ; CALL XREF from sym._init @ 0x8049008                     |
-|   ; CALL XREF from QtPrivate::RefCount::deref() @ 0x8049349  |
-|     ; method.QtPrivate::RefCount.deref                       |
-|   ; CALL XREF from QString::QString(char const*) @ 0x804939d |
-|     ; method.QString.QString_char_const                      |
-|   ; CALL XREF from QString::~QString() @ 0x80493e5           |
-|     ; method.QString._QString                                |
-|   ; CALL XREF from sym._fini @ 0x80495d4                     |
-| sym.__x86.get_pc_thunk.bx ();                                |
-| mov ebx, dword [esp]                                         |
-| ret                                                          |
-`--------------------------------------------------------------'
+.------------------------------------------------------------------.
+|  0x8049140                                                       |
+|   ; CALL XREF from sym._init @ 0x8049008                         |
+|   ; CALL XREF from method.QtPrivate::RefCount.deref @ 0x8049349  |
```

Comment:
Why some `::` are substituted, e.g. in `QString`, but here it doesn't happen?

Reply:
what are you talking about?

Reply:
oh is because the class name is `QtPrivate::RefCount` meanwhile for QString is just `QString`

Reply:
It was `QtPrivate::RefCount::deref()` became `method.QtPrivate::RefCount.deref`, but what was `QString::QString()` became `method.QString.QString_char_const`. Looks inconsistent to me - when it decides to remove `::` and when it doesn't?

Reply:
you are mixing flags with actual demangled strings.
Flags gets created as `"method." + class_name + "." + method` and then the flag is stripped of special chars

Reply:
also as @kazarmy said here https://github.com/rizinorg/rizin/pull/3496#issuecomment-1534118795 the filter trims underscores.

Reply:
Ok, then it's better to sort these things out in a separate PR/whatever, see https://github.com/rizinorg/rizin/issues/3543